src/app/utils/utils.js

```diff
@@ -106,10 +106,36 @@ function Utils() {
    * capitalizes a string
    * @param {string} str
    */
-  this.capitalizeAString = (str) => {
-    return str.replace(/^\w/, function (chr) {
-      return chr.toUpperCase();
+  this.capitalizeAString = (str) =>
+    str.replace(/^\w/, (chr) => chr.toUpperCase());
+
+  /**
+   * focusOnFirstAvailableElement(elementIds)
+   * Jumps the focus to the first available HTML element that is listed in an array of element IDs.
+   * @param {array} elementIds
+   */
+  this.focusOnFirstAvailableElement = (elementIds) => {
+    if (!Array.isArray(elementIds) || !elementIds.length) {
+      return false;
+    }
+
+    elementIds.some((elementId) => {
```

Comment:
Looks to me like this is a `forEach` situation where you want to iterate over `elementIds` array and then test for true/false within the function. `some` seems to be a test in and of itself.

Reply:
Yeah the reason having `some` here is that `forEach` doesn't stop until it has looped every item. So it will always jump focus to the last item. `some` here serves as a way to stop the iteration.

Reply:
For the record, usage of Array#some
https://stackoverflow.com/questions/2641347/how-to-short-circuit-array-foreach-like-calling-break

> This works because some returns true as soon as any of the callbacks, executed in array order, return true, short-circuiting the execution of the rest.

This is more of a learning process for me on understanding the mechanism of Array#some. AFAIK the local instances are working. I think as long as there are comments explaining when the Array#some loop will break, so I don't forget in six months, would be great for me.

Reply:
I can definitely add comments to it.